internal/host/validations_test.go

```diff
@@ -2339,4 +2339,255 @@ var _ = Describe("Validations test", func() {
+			Entry("Machine networks is nil", "192.168.1.1", nil, nil, ValidationPending, "Missing inventory or machine network CIDR"),
+			Entry("Machine networks is empty", "192.168.1.1", []*models.MachineNetwork{}, nil, ValidationPending, "Missing inventory or machine network CIDR"),
+			Entry("Machine networks IPv4 overlaps with host IPv4 address", "192.168.1.1",
+				[]*models.MachineNetwork{
+					{
+						Cidr:      "192.168.1.0/24",
+						ClusterID: clusterID,
+					},
+				},
+				[]*models.Interface{
+					{
+						IPV4Addresses: []string{"192.168.1.1/24"},
+					},
+				},
+				ValidationFailure, "Network interface connected to iSCSI disk cannot belong to machine network CIDRs"),
+			Entry("Machine networks IPv6 overlaps with host IPv6 address", "2001:0db8:85a3::8a2e:0370:7334",
+				[]*models.MachineNetwork{
+					{
+						Cidr:      "2001:0db8:85a3::/48",
+						ClusterID: clusterID,
+					},
+				},
+				[]*models.Interface{
+					{
+						IPV6Addresses: []string{"2001:0db8:85a3::8a2e:0370:7334/48"},
+					},
+				},
+				ValidationFailure, "Network interface connected to iSCSI disk cannot belong to machine network CIDRs"),
+			Entry("Machine networks IPv6 uses the same interface as host IPv4 address", "192.168.1.1",
+				[]*models.MachineNetwork{
+					{
+						Cidr:      "2001:0db8:85a3::/48",
+						ClusterID: clusterID,
+					},
+				},
+				[]*models.Interface{
+					{
+						IPV4Addresses: []string{"192.168.1.1/24"},
+						IPV6Addresses: []string{"2001:0db8:85a3::8a2e:0370:7334/48"},
+					},
+				},
+				ValidationFailure, "Network interface connected to iSCSI disk cannot belong to machine network CIDRs"),
+			Entry("Machine networks IPv4 uses the same interface as host IPv6 address", "2001:0db8:85a3::8a2e:0370:7334",
+				[]*models.MachineNetwork{
+					{
+						Cidr:      "192.168.1.1/24",
+						ClusterID: clusterID,
+					},
+				},
+				[]*models.Interface{
+					{
+						IPV4Addresses: []string{"192.168.1.1/24"},
+						IPV6Addresses: []string{"2001:0db8:85a3::8a2e:0370:7334/48"},
+					},
+				},
+				ValidationFailure, "Network interface connected to iSCSI disk cannot belong to machine network CIDRs"),
+			Entry("Machine networks IPv4 and host IPv4 address use different interfaces", "192.168.1.1",
+				[]*models.MachineNetwork{
+					{
+						Cidr:      "192.168.2.1/24",
+						ClusterID: clusterID,
+					},
+				},
+				[]*models.Interface{
+					{
+						IPV4Addresses: []string{"192.168.1.1/24"},
+					},
+					{
+						IPV4Addresses: []string{"192.168.2.1/24"},
+					},
+				},
+				ValidationSuccess, "Network interface connected to iSCSI disk does not belong to machine network CIDRs"),
+			Entry("Machine networks IPv4/IPv6 and host IPv4 address use different interfaces", "192.168.1.1",
+				[]*models.MachineNetwork{
+					{
+						Cidr:      "192.168.2.1/24",
+						ClusterID: clusterID,
+					},
+					{
+						Cidr:      "2001:0db8:85a3::/48",
+						ClusterID: clusterID,
+					},
+				},
+				[]*models.Interface{
```

Comment:
Is it worth adding a test case with an unparseable IP prefix? For both NIC and machine network?

Reply:
Actually if test separately and additionally only the function it will be way easier, and we could add all these edge cases

Reply:
MachineNetwork is hard to test because swagger already validates that the field is already a CIDR. So the test don't even comes to this validation.

Reply:
agree that if I may extract a unit test if I split the function